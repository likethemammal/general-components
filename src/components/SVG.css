.container {
    max-width: 100%;
    max-height: 100%;
}

.container svg {
    fill: white;
    width: 100%;
    height: 100%;
}