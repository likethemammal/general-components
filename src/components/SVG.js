import React, { Component } from 'react'

import css from './SVG.css'

const SVG_PREFIX = 'data:image/svg+xml;utf8,'

export default class SVG extends Component {

    static defaultProps = {
        className: css.container,
    }

    render() {

        const { raw, className } = this.props

        return <div
            className={className}
            dangerouslySetInnerHTML={{__html: raw}}
        />
    }
}