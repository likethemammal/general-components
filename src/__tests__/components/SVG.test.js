import React from 'react'
import {shallow} from 'enzyme'

import * as shared from '../_shared/tests'
import SVG from '../../components/SVG'


const component_jsx = <SVG />

describe('SVG', () => {

    const component = shallow(
        component_jsx,
    )

    shared.SNAPSHOTS_SHOULD_MATCH([
        component,
    ])
})