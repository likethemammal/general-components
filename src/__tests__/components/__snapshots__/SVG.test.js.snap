// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SVG Snapshot 0 should match 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": undefined,
    }
  }
/>
`;
